datadog_checks_dev/tests/test_kind.py

```diff
@@ -0,0 +1,50 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import mock
+import pytest
+import tenacity
+
+from datadog_checks.dev import RetryError
+from datadog_checks.dev.ci import running_on_ci
+from datadog_checks.dev.kind import kind_run
+
+from .common import not_windows_ci
+
+
+class TestKindRun:
+    @pytest.mark.parametrize(
+        "attempts,expected_call_count",
+        [
+            (None, 1),
+            (0, 1),
+            (1, 1),
+            (3, 3),
+        ],
+    )
+    @not_windows_ci
+    def test_retry_on_failed_conditions(self, attempts, expected_call_count):
+        condition = mock.MagicMock()
+        condition.side_effect = RetryError("error")
```

Comment:
If I'm understanding this correctly, this could be any error. Maybe we can pick something different, to avoid confusion with `tenacity.RetryError`.

Reply:
Very true. Modified & see my other comment.